sssd_test_framework/utils/sssd.py

```diff
@@ -783,3 +784,49 @@ def autofs(self) -> None:
         """
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
+
+    def proxy(
+        self,
+        proxy: Literal["files", "ldap"],
+        provider: str | list[str] = "id",
+        proxy_file: str = """""",
+        sever_hostname: str = """""",
+        domain: str | None = None,
+    ):
+        """
+        :param proxy: ``ldap`` or ``files``
+        :type proxy: str
+        :param provider: SSSD providers (id, auth, chpass, ...)
+        :type provider: str | list[str]
+        :param proxy_file: Config for proxy
+        :type proxy_file: str
+        :param sever_hostname: LDAP (LDAP server hostname)
+        :type sever_hostname: str
+        :param domain: Existing domain name, defaults to None (= default domain)
+        :type domain: str | None, optional
+        """
+        if domain is None:
+            domain = self.sssd.default_domain
+        assert domain is not None
```

Comment:
Like this rest of the code in this file, if there isn't a domain, let's raise an exception. 

        if domain is None:
            raise ValueError("No domain specified!")